### DISCUSSION TITLE
Powell lens

### DISCUSSION THREAD

**Discussion by ppfromero** [mjhoptics/ray-optics#112] [Q&A] [answered]
Hi guys,

I'm trying to learn how to design powell lens, to render gaussian beam to top hat beam.
Do you have any example to to build this kind of lens, any ideas or documents to share?

Regards

José

**Comment by mjhoptics** [comment, marked as answer]
Hello Jose,
I don't know much about Powell lenses but it appears that they require
non-sequential ray tracing to model them correctly. ray-optics is a
sequential ray tracing package so it doesn't handle that scenario currently.
Regards
Mike Hayford

**Reply by ppfromero** [reply]
Hello Mike!
Thanks for your answer!
I continue to search a package that can mange this!!
Regards
Jose
